src/lib/queries/ibc.ts

```diff
@@ -207,6 +237,22 @@ export function setupIbcExtension(base: QueryClient): IbcExtension {
           channelQueryService.Channels({
             pagination: createPagination(paginationKey),
           }),
+        allChannels: async () => {
+          const channels = [];
+          let response: QueryChannelsResponse;
+          let key: Uint8Array | undefined;
+          do {
+            response = await channelQueryService.Channels({
+              pagination: createPagination(key),
+            });
+            channels.push(...response.channels);
+            key = response.pagination?.nextKey;
+          } while (key);
```

Comment:
Hehe, it's been years since I saw a good use of `do...while`. This is one.

Reply:
I needed one the other day in some Python code so it was nice to be able to actually use it here!